L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp3.cc

```diff
@@ -956,18 +964,29 @@ void HGCalTriggerGeometryV9Imp3::etaphiMappingFromSector0(int& ieta, int& iphi,
 
 HGCalGeomRotation::WaferCentring HGCalTriggerGeometryV9Imp3::getWaferCentring(unsigned layer, int subdet) const {
   if (subdet == HGCalTriggerSubdetector::HGCalEETrigger) {  // CE-E
-    return HGCalGeomRotation::WaferCentring::WaferCentred;
+    return HGCalGeomRotation::WaferCentring::WaferCentred;  // All layers are wafer centred
   } else if (subdet == HGCalTriggerSubdetector::HGCalHSiTrigger) {
+    auto topologyLayerType = hsiTopology().dddConstants().layerType(layer);
+
     if ((layer % 2) == 1) {  // CE-H Odd
-      return HGCalGeomRotation::WaferCentring::CornerCentredY;
+      // Check topology for wafer centred layers, otherwise CornerCentredY
+      if (topologyLayerType == HGCalTypes::WaferCenter || topologyLayerType == HGCalTypes::WaferCenterR ||
+          topologyLayerType == HGCalTypes::WaferCenterB)
+        return HGCalGeomRotation::WaferCentring::WaferCentred;
+      else
+        return HGCalGeomRotation::WaferCentring::CornerCentredY;
     } else {  // CE-H Even
-      return HGCalGeomRotation::WaferCentring::CornerCentredMercedes;
+      // Check topology for wafer centred layers, otherwise CornerCentredMercedes
+      if (topologyLayerType == HGCalTypes::WaferCenter || topologyLayerType == HGCalTypes::WaferCenterR ||
+          topologyLayerType == HGCalTypes::WaferCenterB)
+        return HGCalGeomRotation::WaferCentring::WaferCentred;
+      else
+        return HGCalGeomRotation::WaferCentring::CornerCentredMercedes;
```

Comment:
Do we still need to separate between odd and even layers if we can retrieve the type of layer centring with `dddConstants.layerType()`?

Reply:
Looking at where `dddConstants.layerType()` retrieves the layer type:

https://github.com/hgc-tpg/cmssw/blob/7724d3db1bab4131b04e142130aa0d7dc420df4b/Geometry/HGCalCommonData/src/HGCalTypes.cc#L42-L48

If `HGCalTypes::CornerCenterYp` and `HGCalTypes::CornerCenterYm` uniquely correspond to one of `HGCalGeomRotation::WaferCentring::CornerCentredY` and `HGCalGeomRotation::WaferCentring::CornerCentredMercedes`, then I think we can remove the separation between odd and even layers.  Though I don't know which of `CornerCenterYp/m` corresponds to `CornerCentredY/Mercedes` - I can work this out, unless you happen to know.

Reply:
I would naively guess that `Yp` is `Y` and `Ym` is `Mercedes` (negative version of `Y`), but not sure at all.

Reply:
This is what I would have guessed, but from trying it out, it seems to be the other way round.  With this info, I've made a commit to retrieve the wafer centring of CE-H layers based only on `dddConstants.layerType()`.